### PR TITLE
[W3M] Add web3modal api integration

### PR DESCRIPTION
**Pull request by kacperoak**

### CODE REVIEW

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/Web3ModalViewModel.kt [hunk 1]
```diff
@@ -123,18 +121,18 @@ internal class Web3ModalViewModel : ViewModel() {
...
-            val wallets = if (Web3Modal.recommendedWalletsIds.isEmpty()) {
-                getWalletsUseCase(sdkType = W3M_SDK, chains = chains, excludedIds = Web3Modal.excludedWalletsIds)
-            } else {
-                getWalletsUseCase(sdkType = W3M_SDK, chains = chains, excludedIds = Web3Modal.excludedWalletsIds, recommendedIds = Web3Modal.recommendedWalletsIds).union(
-                    getWalletsUseCase(sdkType = W3M_SDK, chains = chains, excludedIds = Web3Modal.excludedWalletsIds)
-                ).toList()
+        viewModelScope.launch {
+            try {
+                val wallets = getAllWalletsUseCase(
+                    sdkType = "w3m",
```
Comment (Elyniss) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/Web3ModalViewModel.kt:127]: Why is this no longer constant?

File: sample/modals/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+        <package android:name="io.zerion.android"/>
+        <package android:name="com.spot.spot"/>
+        <package android:name="fi.steakwallet.app"/>
+        <package android:name="vip.mytokenpocket"/>
+        <package android:name="com.frontierwallet"/>
+        <package android:name="com.bitkeep.wallet"/>
+        <package android:name="im.argent.contractwalletclient"/>
+        <package android:name="com.walletconnect.sample.wallet"/>
+    </queries>
+
```
Comment (TalhaAli00) [inline sample/modals/src/main/AndroidManifest.xml:19]: is this the complete list? I believe RN has something close to 40 packages that it adds to the manifest
https://github.com/WalletConnect/react-native-examples/blob/1529f010475603c10260fd14138b73f358e48c4c/dapps/v2Explorer/android/app/src/main/AndroidManifest.xml#L46

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/redirect/RedirectWalletScreen.kt [hunk 3]
```diff
@@ -113,10 +113,12 @@ private fun RedirectWalletScreen(
             onClick = onCopyLinkClick
         )
         VerticalSpacer(height = 16.dp)
-        FullWidthDivider()
-        VerticalSpacer(height = 16.dp)
-        StoreEntry(text = "Get ${wallet.name}", onClick = onOpenPlayStore)
-        VerticalSpacer(height = 16.dp)
+        if (wallet.playStore != null) {
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/redirect/RedirectWalletScreen.kt:116]: maybe the `VerticalSpacer` should also be included in the if

File: core/android/src/main/kotlin/com/walletconnect/android/internal/common/modal/domain/usecase/GetAllWalletsUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.walletconnect.android.internal.common.modal.domain.usecase
+
+import com.walletconnect.android.internal.common.modal.Web3ModalApiRepository
+import com.walletconnect.android.internal.common.modal.data.model.Wallet
+
+class GetAllWalletsUseCase(
```
Comment (TalhaAli00) [inline core/android/src/main/kotlin/com/walletconnect/android/internal/common/modal/domain/usecase/GetAllWalletsUseCase.kt:6]: if you add an interface for this, then you can make the concrete class internal. There's an example in Notify.engine for reference

File: core/android/src/main/kotlin/com/walletconnect/android/internal/common/modal/Web3ModalApiRepository.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package com.walletconnect.android.internal.common.modal
+
+import android.content.Context
+import com.walletconnect.android.internal.common.modal.data.model.Wallet
+import com.walletconnect.android.internal.common.modal.data.network.Web3ModalService
+import com.walletconnect.android.internal.common.modal.data.network.model.WalletDTO
+import com.walletconnect.android.utils.isWalletInstalled
+
+class Web3ModalApiRepository(
```
Comment (TalhaAli00) [inline core/android/src/main/kotlin/com/walletconnect/android/internal/common/modal/Web3ModalApiRepository.kt:9]: internal

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**